### PR TITLE
refactor: adjust process definition domain entities

### PR DESCRIPTION
**Pull request by houssain-barouni**
## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

same as https://github.com/camunda/camunda/pull/24750
relates to https://github.com/camunda/camunda/issues/24911

### CODE REVIEW

**Review by stoerti — APPROVED** [review]
LGTM 👍

File: clients/java/src/main/java/io/camunda/zeebe/client/ZeebeClient.java [hunk 1]
```diff
@@ -1033,13 +1033,13 @@ static ZeebeClientCloudBuilderStep1 newCloudClientBuilder() {
   DecisionInstanceQuery newDecisionInstanceQuery();
 
   /**
-   * Retrieves a decision instance by key.
+   * Retrieves a decision instance by id.
    *
    * <pre>
-   * long decisionInstanceKey = ...;
+   * String decisionInstanceKey = ...;
```
Comment (tmetzke) [inline clients/java/src/main/java/io/camunda/zeebe/client/ZeebeClient.java:1039]:
```suggestion
   * String decisionInstanceId = ...;
```

File: qa/integration-tests/src/test/java/io/camunda/it/client/DecisionInstanceQueryTest.java [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+              DECISION_DEFINITION_ID_2,
+              "{\"amount\": 100, \"invoiceCategory\": \"Misc\"}"));
+      waitForDecisionsToBeEvaluated(
+          zeebeClient,
+          evaluatedDecisions.values().stream()
+              .mapToInt(v -> v.getEvaluatedDecisions().size())
+              .sum());
+      initialized = true;
+    }
+  }
```
Comment (tmetzke) [inline qa/integration-tests/src/test/java/io/camunda/it/client/DecisionInstanceQueryTest.java:71]: ❓ Is there anything that prevents us from doing a `BeforeAll` which is executed once for the test class? We are now effectively running this once for the test class, right?

File: qa/integration-tests/src/test/java/io/camunda/it/client/DecisionInstanceQueryTest.java [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+
+    // when
+    final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSXX");
+    final var offsetDateTime = OffsetDateTime.parse(di.getEvaluationDate(), formatter);
+    final var result =
+        zeebeClient
+            .newDecisionInstanceQuery()
+            .filter(f -> f.evaluationDate(offsetDateTime.toString()))
+            .send()
+            .join();
```
Comment (tmetzke) [inline qa/integration-tests/src/test/java/io/camunda/it/client/DecisionInstanceQueryTest.java:187]: 💭 By the way, @danielkelemen is currently building [this here](https://github.com/camunda/camunda/pull/24760), which should make it a lot nicer to write this 😃

**Review by tmetzke — APPROVED** [review]
👍 Looks good overall, just some minor suggestions 😃

**Comment by houssain-barouni** [comment]
@tmetzke the diff does not reflect the reality, I need to rebase the branch

File: qa/integration-tests/src/test/java/io/camunda/it/client/DecisionInstanceQueryTest.java (hunk 2 above)
Comment (houssain-barouni) [reply qa/integration-tests/src/test/java/io/camunda/it/client/DecisionInstanceQueryTest.java]: I tried to use @BeforeAll but could not have the parameter resolver working 
`org.junit.jupiter.api.extension.ParameterResolutionException: No ParameterResolver registered for parameter [final io.camunda.zeebe.client.ZeebeClient zeebeClient] in method ...`
It seems that [this](https://github.com/camunda/camunda/blob/1753c1efd1a5480847d411f19c77f5536e06cf96/qa/integration-tests/src/test/java/io/camunda/it/utils/BrokerWithCamundaExporterITInvocationProvider.java#L147) is invoked afterward.
Probably, `BeforeAll` is not compatible with ParameterResolver used within the context of `@TestTemplate`.

I haven’t examined this in detail, but I prefer not to spend too much time on it.

**Pull request merged**